cumulus/parachains/runtimes/collectives/collectives-westend/src/impls.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::OriginCaller;
+use frame_support::{
+	dispatch::DispatchResultWithPostInfo,
+	traits::{Currency, Get, Imbalance, OnUnbalanced, OriginTrait, PrivilegeCmp},
+	weights::Weight,
+};
+use log;
+use pallet_alliance::{ProposalIndex, ProposalProvider};
+use parachains_common::impls::NegativeImbalance;
+use sp_runtime::DispatchError;
+use sp_std::{cmp::Ordering, marker::PhantomData, prelude::*};
+use xcm::latest::{Fungibility, Junction, Parent};
+
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+type ProposalOf<T, I> = <T as pallet_collective::Config<I>>::Proposal;
+
+type HashOf<T> = <T as frame_system::Config>::Hash;
+
+/// Type alias to conveniently refer to the `Currency::Balance` associated type.
+pub type BalanceOf<T> =
+	<pallet_balances::Pallet<T> as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// Implements `OnUnbalanced::on_unbalanced` to teleport slashed assets to relay chain treasury
+/// account.
+pub struct ToParentTreasury<TreasuryAccount, PalletAccount, T>(
+	PhantomData<(TreasuryAccount, PalletAccount, T)>,
+);
+
+impl<TreasuryAccount, PalletAccount, T> OnUnbalanced<NegativeImbalance<T>>
+	for ToParentTreasury<TreasuryAccount, PalletAccount, T>
```

Comment:
Sounds like this could be a more generally usable type in one of the common crates.

Reply:
Turns out that @joepetrowski has added this [here](https://github.com/paritytech/polkadot-sdk/pull/2281/commits/fa4e4e60f9c207cb5b5cc6a54139cb7c958c0533) and if this PR is merged soon, he will edit this in his PR to use the type he has added.
If not I'll make a follow-up PR to not block his.